tools/pepsi/src/cargo.rs

```diff
@@ -1,151 +1,215 @@
-            .await
-            .wrap_err("failed to run clippy")?,
-        Command::Run => {
-            if arguments.workspace {
-                println!("INFO: Found --workspace with run subcommand, ignoring...")
-            }
-            repository
-                .run(
-                    &arguments.profile,
-                    &arguments.target,
-                    arguments.features,
-                    &arguments.passthrough_arguments,
-                )
-                .await
-                .wrap_err("failed to run")?
-        }
+    arguments.cargo.apply(&mut cargo);
+
+    let mut cargo_command = cargo
+        .command(repository, compiler_artifacts)
+        .wrap_err("failed to create cargo command")?;
+
+    debug!("Running `{cargo_command:?}`");
+
+    let status = cargo_command
+        .status()
+        .await
+        .wrap_err("failed to run cargo")?;
+
+    if !status.success() {
+        bail!("cargo failed with {status}");
     }
 
     Ok(())
 }
 
-pub async fn remote(arguments: Arguments, command: Command) -> Result<()> {
-    match command {
-        Command::Build => {
-            let mut command = TokioCommand::new("./scripts/remoteWorkspace");
-
-            let profile_name = match arguments.profile.as_str() {
-                "dev" => "debug",
-                other => other,
-            };
-            let toolchain_name = match arguments.target.as_str() {
-                "nao" => "x86_64-aldebaran-linux-gnu/",
-                _ => "",
-            };
-            command.args([
-                "--return-file",
-                &format!(
-                    "target/{toolchain_name}{profile_name}/hulk_{}",
-                    arguments.target
-                ),
-            ]);
-
-            command
-                .arg("./pepsi")
-                .arg("build")
-                .arg("--profile")
-                .arg(arguments.profile)
-                .arg("--target")
-                .arg(arguments.target);
-
-            if arguments.workspace {
-                command.arg("--workspace");
-            }
-            if arguments.no_sdk_installation {
-                command.arg("--no-sdk-installation");
-            }
-            command.arg("--");
-            command.args(arguments.passthrough_arguments);
+async fn read_requested_environment(manifest_path: &Option<PathBuf>) -> Result<Environment> {
+    let Some(manifest_path) = manifest_path else {
+        return Ok(Environment::Native);
+    };
 
-            let status = command
-                .status()
-                .await
-                .wrap_err("failed to execute remote script")?;
+    let manifest = read_to_string(manifest_path).await.wrap_err_with(|| {
+        format!(
+            "failed to read manifest at {path}",
+            path = manifest_path.display()
+        )
+    })?;
+    let manifest: Table = toml::from_str(&manifest).wrap_err("failed to parse manifest")?;
```

Comment:
Add `path` to error message like above.

Reply:
done

Reply:
[I don't see it](https://github.com/schmidma/hulk/blob/pepsi-max/tools/pepsi/src/cargo.rs#L144).